purge-docker-cluster.yml

```diff
@@ -439,28 +463,32 @@
       ansible_pkg_mgr == "dnf" and
       not is_atomic
 
-  - name: remove six
-    pip:
-      name: six
-      version: 1.9.0
-      state: absent
-    when: not is_atomic
+  - name: remove package dependencies on redhat
+    command: yum -y autoremove
+    when:
+      ansible_os_family == 'RedHat' and
+      ansible_pkg_mgr == "yum" and
+      not is_atomic
 
-  - name: remove docker-py
-    pip:
-      name: docker-py
-      version: 1.1.0
-      state: absent
+  - name: remove package dependencies on redhat again
+    command: yum -y autoremove
     when:
-      ansible_version['full'] | version_compare('2.1.0.0', '<') and
+      ansible_os_family == 'RedHat' and
+      ansible_pkg_mgr == "yum" and
       not is_atomic
 
-  - name: remove docker-py
-    pip:
-      name: docker-py
-      state: absent
+  - name: remove package dependencies on redhat
+    command: dnf -y autoremove
```

Comment:
That's because the dnf module does not support the autoremove command. I am going to file a feature request on ansible for that.